paper/examples/flagstat/flagstat.go

```diff
@@ -30,17 +31,26 @@ func main() {
 		log.Fatal(err)
 	}
 	defer f.Close()
+	ok, err := bgzf.HasEOF(f)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if !ok {
+		log.Println("EOF block missing")
+	}
+
 	b, err := bam.NewReader(f, 0)
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer b.Close()
-	b.Omit(2)
+	b.Omit(bam.AllVariableLengthData)
 
 	// counts is indexed by [pass/fail][sam.Flag] where we have 12 possible sam Flags.
 	var counts [2][12]uint64
 	// track mates on different chromosomes.
-	var mates [2]struct{ low, high uint64 }
+	var mates [2]struct{ allMapQ, highMapQ uint64 }
+	var good, singletons, paired [2]uint64
 	var qc int
 	for {
 		read, err := b.Read()
```

Comment:
Yeah, it's unlikely that this is causing that. The value doesn't escape to the heap, so it's not an expensive allocation.